src/chains/index.ts

```diff
@@ -8,62 +8,93 @@ import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
 import { IS_TEST } from '@/env';
 import buildTimeNetworks from '@/references/networks.json';
 
-// NOTE: Prefer runtime data from backendNetworksQueryKey, but fallback to buildTimeNetworks if needed
-const backendNetworks = queryClient.getQueryData<BackendNetworksResponse>(backendNetworksQueryKey()) ?? buildTimeNetworks;
-
-const BACKEND_CHAINS = transformBackendNetworksToChains(backendNetworks.networks);
-
-export const SUPPORTED_CHAINS: Chain[] = IS_TEST ? [...BACKEND_CHAINS, chainHardhat, chainHardhatOptimism] : BACKEND_CHAINS;
-
-export const defaultChains: Record<ChainId, Chain> = SUPPORTED_CHAINS.reduce(
-  (acc, chain) => {
-    acc[chain.id] = chain;
-    return acc;
-  },
-  {} as Record<ChainId, Chain>
-);
-
-export const SUPPORTED_CHAIN_IDS = SUPPORTED_CHAINS.map(chain => chain.id);
-
-export const SUPPORTED_MAINNET_CHAINS: Chain[] = SUPPORTED_CHAINS.filter(chain => !chain.testnet);
-
-export const SUPPORTED_MAINNET_CHAIN_IDS: ChainId[] = SUPPORTED_MAINNET_CHAINS.map(chain => chain.id);
-
-export const needsL1SecurityFeeChains = backendNetworks.networks
-  .filter((backendNetwork: BackendNetwork) => backendNetwork.opStack)
-  .map((backendNetwork: BackendNetwork) => parseInt(backendNetwork.id, 10));
-
-export const chainsNativeAsset: Record<number, BackendNetwork['nativeAsset']> = backendNetworks.networks.reduce(
-  (acc, backendNetwork: BackendNetwork) => {
-    acc[parseInt(backendNetwork.id, 10)] = backendNetwork.nativeAsset;
-    return acc;
-  },
-  {} as Record<number, BackendNetwork['nativeAsset']>
-);
-
-export const chainsLabel: Record<number, string> = backendNetworks.networks.reduce(
-  (acc, backendNetwork: BackendNetwork) => {
-    acc[parseInt(backendNetwork.id, 10)] = backendNetwork.label;
-    return acc;
-  },
-  {} as Record<number, string>
-);
-
-export const chainsName: Record<number, string> = backendNetworks.networks.reduce(
-  (acc, backendNetwork: BackendNetwork) => {
-    acc[parseInt(backendNetwork.id, 10)] = backendNetwork.name;
-    return acc;
-  },
-  {} as Record<number, string>
-);
-
-export const chainsIdByName: Record<string, number> = backendNetworks.networks.reduce(
-  (acc, backendNetwork: BackendNetwork) => {
-    acc[backendNetwork.name] = parseInt(backendNetwork.id, 10);
-    return acc;
-  },
-  {} as Record<string, number>
-);
+const getBackendNetworks = (): BackendNetworksResponse => {
+  return queryClient.getQueryData<BackendNetworksResponse>(backendNetworksQueryKey()) ?? buildTimeNetworks;
```

Comment:
This won't work for parts of the app that require UI thread execution, specifically https://github.com/rainbow-me/rainbow/blob/5d5dfed90bbef1b9b39534c7528f09f095997a3d/src/__swaps__/screens/Swap/components/TokenList/ChainSelection.tsx#L51